### DISCUSSION TITLE
DataGrid Footer

### DISCUSSION THREAD

**Discussion by mrcorale** [Megabit/Blazorise#3404] [General]
Hi, I have a problem with Footer on Datagrid, I can't add Total to the Amount!

                                <Table>
                                    <TableHeader ThemeContrast="ThemeContrast.Light">
                                        <DataGrid TItem="CascoRate"
                                                  Data="@objectRate"
                                                  Sortable="false"
                                                  Filterable="false"
                                                  Editable="false"
                                                  Hoverable="false"
                                                  PageSize="5"
                                                  ShowPager="true"
                                                  Narrow="true"
                                                  Bordered="true"
                                                  UseInternalEditing="false"
                                                  RowClicked="OnClickRate"
                                                  RowDoubleClicked="DoubleClickRate"
                                                  SelectedRowStyling="@OnSelectRowStyleRate"
                                                  EditMode="DataGridEditMode.Inline"
                                                  RowUpdated="@SaveModalRate"
                                                  RowRemoved="@RemoveRowRate"
                                                  @bind-SelectedRow="@Rate">

                                            <FirstPageButtonTemplate><span class="stpglst">Prima</span></FirstPageButtonTemplate>
                                            <NextPageButtonTemplate><span class="stpglst">Înainte</span></NextPageButtonTemplate>
                                            <PreviousPageButtonTemplate>
                                                <span class="stpglst">Înapoi</span>

                                                <DataGridColumn TItem="CascoRate" Field="@nameof(Rate.DataRata)" Width="90px" Caption="Data rată" DisplayFormat="{0:dd.MM.yyyy}" Sortable="false" Editable="true" />
                                                <DataGridNumericColumn TItem="CascoRate" Field="@nameof(Rate.SumaRata)" Width="100px" Caption="Suma rată" Sortable="false" Editable="true" />
                                            </PreviousPageButtonTemplate>
                                            <LastPageButtonTemplate><span class="stpglst">Ultima</span></LastPageButtonTemplate>
                                        </DataGrid>

                                        <Alert Color="Color.Danger" @bind-Visible="@visibleObRateEror">
                                            <AlertMessage>@mesajObRateEror</AlertMessage>
                                            <CloseButton Clicked="@(() => visibleObRateEror = false)" />
                                        </Alert>

                                        <Modal @ref="modalRate">
                                            <ModalContent Class="stmodobS" Centered="true">
                                                <ModalHeader>
                                                    <ModalTitle Class="stylemodaltitle">Casco Rate</ModalTitle>
                                                    <CloseButton Clicked="@HideModalRate" />
                                                </ModalHeader>
                                                <ModalBody>
                                                    <Fields>
                                                        <Field>
                                                            <FieldLabel Class="sttextfield">Data rată</FieldLabel>
                                                            <FieldBody>
                                                                <DateEdit TValue="DateTime?" Date="@DataRata" DateChanged="@OndDatRatChanged" />
                                                            </FieldBody>
                                                        </Field>
                                                    </Fields>

                                                    <Fields>
                                                        <Field>
                                                            <FieldLabel Class="sttextfield">Suma rate</FieldLabel>
                                                            <FieldBody>
                                                                <Validation Validator="SumaRataValidate">
                                                                    <NumericEdit TValue="decimal?" @bind-Value="@SumaRata" Placeholder="adaugă suma rată..." />
                                                                    <ValidationSuccess></ValidationSuccess>
                                                                    <ValidationError></ValidationError>
                                                                </Validation>
                                                            </FieldBody>
                                                        </Field>
                                                    </Fields>
                                                </ModalBody>
                                                <ModalFooter>
                                                    <Button Class="stbtnall" @onclick="() => SaveModalRate()">OK</Button>
                                                    <Button Class="stbtnall" Margin="Margin.Is1.FromRight" Clicked="() => HideModalRate()">Cancel</Button>
                                                </ModalFooter>
                                            </ModalContent>
                                        </Modal>
                                    </TableHeader>
                                </Table>

**Comment by David-Moreira** [comment]
Hello @mrcorale, 
Can you elaborate on that? What are you trying to accomplish? can you add the `@code` portion also, preferably with the class definitions and such?

**Reply by mrcorale** [reply]
this is the code I created, for business logic, I don't want to declare the code and conditions!

**Reply by stsrki** [reply]
There are several problems with your code. First, you have DataGrid(along with all other components) placed in a TableHeader. That means you have a table in a table. And semantically based on HTML specs fr table that is problematic.

To add a footer to a DataGrid you need to use aggregates. See the example in our docs https://blazorise.com/docs/extensions/datagrid/aggregates

**Reply by mrcorale** [reply]
Thank you very much, I understand what needs to be done !

**Reply by stsrki** [reply]
Can you just mark the ticket as answered? tnx
